apps/dashboard/src/components/variable/components/reorder-filter-item.tsx

```diff
@@ -1,80 +1,182 @@
-import { Button } from '@/components/primitives/button';
+import { ComponentProps, useMemo, useRef } from 'react';
+import { Code2, GripVertical } from 'lucide-react';
+import { Reorder, useDragControls, useMotionValue } from 'motion/react';
+import { RiCloseLine, RiQuestionLine } from 'react-icons/ri';
+import { FeatureFlagsKeysEnum } from '@novu/shared';
+
+import { buttonVariants } from '@/components/primitives/button';
 import { FilterWithParam } from '../types';
-import { InputPure } from '@/components/primitives/input';
+import { Input } from '@/components/primitives/input';
 import { Tooltip, TooltipContent, TooltipTrigger } from '@/components/primitives/tooltip';
-import { GripVertical } from 'lucide-react';
-import { Reorder } from 'motion/react';
-import { ComponentProps, useMemo } from 'react';
-import { RiCloseLine } from 'react-icons/ri';
 import { getFilters } from '../constants';
 import { useFeatureFlag } from '@/hooks/use-feature-flag';
-import { FeatureFlagsKeysEnum } from '@novu/shared';
+import { AccordionContent } from '@/components/primitives/accordion';
+import { AccordionTrigger } from '@/components/primitives/accordion';
+import { AccordionItem } from '@/components/primitives/accordion';
+import { Accordion } from '@/components/primitives/accordion';
+import { cn } from '@/utils/ui';
+import { VariableSelect } from '@/components/conditions-editor/variable-select';
+import { LiquidVariable } from '@/utils/parseStepVariables';
+
+const preventClick = (e: React.MouseEvent) => {
+  e.stopPropagation();
+  e.preventDefault();
+};
 
 type ReorderFilterItemProps = ComponentProps<typeof Reorder.Item<FilterWithParam>> & {
   index: number;
   isLast: boolean;
+  variableName: string;
+  variables: LiquidVariable[];
   onRemove: (value: string) => void;
   onParamChange: (index: number, params: string[]) => void;
 };
 
 export const ReorderFilterItem = (props: ReorderFilterItemProps) => {
-  const { index, isLast, onRemove, onParamChange, value, ...rest } = props;
+  const controls = useDragControls();
+  const x = useMotionValue(0);
+  const { index, isLast, onRemove, onParamChange, value, variableName, variables, ...rest } = props;
   const isEnhancedDigestEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_ENHANCED_DIGEST_ENABLED);
   const liquidFilters = useMemo(() => getFilters(isEnhancedDigestEnabled), [isEnhancedDigestEnabled]);
+  const itemRef = useRef<HTMLDivElement>(null);
 
   const filterDef = liquidFilters.find((t) => t.value === value.value);
+  const hasParams = filterDef?.hasParam && filterDef.params;
+
+  const options = useMemo(() => {
+    // if it's digest step events variable then fill the options with the payload variables
+    if (variableName.match(/^steps\..+\.events$/)) {
+      return variables.filter((v) => v.name.startsWith('payload')).map((v) => ({ label: v.name, value: v.name }));
+    }
+
+    return [];
+  }, [variableName, variables]);
```

Comment:
Create the options for the `toSentence` filter key path component when it's used under the `steps.x.events` variable.